src/main/java/com/google/devtools/build/bfg/bfg.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto2";
+
+package com.google.devtools.build.bfg;
+
+option java_package = "protos.com.google.devtools.build.bfg";
+
+// ParserOutput is used to communicate between language-specific parsers and the language-agnostic
+// BFG core.
+message ParserOutput {
+    // The class dependency graph.
+    // In the Java case, we have (u, v) if 'u' mentions [1] 'v'.
+    // [1] - By mention, we mean either imports or references in code, either as a fully qualified
+    //       name, or an implied same-package name.
+    map<string, Strings> class_to_class = 1;
```

Comment:
can you rename to `class_dependencies`. 
`getClassDependencyMap()` is much clearer than `getClassToClassMap()`

Reply:
"dependencies" can mean several things, I think that "class to class" describes it more precisely. 
There are soon going to be additional fields (class_to_file, file_to_rule_kind) which will be consistent with the new name.
What do you think?

Reply:
> There are soon going to be additional fields (class_to_file, file_to_rule_kind) which will be consistent with the new name.

This is the portion that is unclear to me, the future direction. Can you expand on when and how these additional fields would be used? How would you obtain these file_to_rule dependencies? 

That aside, from my limited vantage point, these are all dependency graphs. The nodes, (i.e. type of dependency) may differ but at their core, they are dependencies. For example, an edge between a file and another rule indicates a dependency between that file and that rule. Consequently, I am curious why not define something like
```
message ParserOutput {
    map<Deps, Deps> dependency_map = 1;
}

message Dep {
     string name;
     string type;
}

message Deps {
    repeated Dep deps;
}
```

Reply:
Note, the main takeaway you should get out of my message is _not_ the specific proto I defined. It's that I don't understand what the direction is.

Reply:
I concede on this